kube-derive/src/custom_resource.rs

```diff
@@ -64,6 +66,42 @@ fn default_serde_json() -> Path {
     parse_quote! { ::serde_json }
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+enum SchemaMode {
+    Disabled,
+    Custom,
+    Derived,
+}
+
+impl SchemaMode {
+    fn derive(self) -> bool {
+        match self {
+            SchemaMode::Disabled => false,
+            SchemaMode::Custom => false,
+            SchemaMode::Derived => true,
+        }
```

Comment:
`matches!(SchemaMode::Derived)`

Reply:
Those were intentional, to force anyone adding any new cases to also consider what happens in these cases.

Reply:
Fair. I was under the impression that the disabled one would go away, but it seems like it is useful for non-owned resources.